sherpa/tests/test_fit_unit.py

```diff
@@ -2534,3 +2534,267 @@ def test_wstat_rstat_qval_fields_not_none():
+    mdl.c2.thaw()
+    mdl.c3.thaw()
+    mdl.c0 = 5.1
+
+    fit = Fit(d, mdl, stat=stat(), method=method())
+    statinfo = fit.calc_stat_info()
+    assert_stat_info(statinfo, 3, -1, statargs[0], statargs[1], statargs[2])
+
+
+# Results from CIAO 4.10 on Linux Ubuntu. Expect to end up with
+# c0 = 3.0, c1 = 1.0 to about 3dp and the rstat values, when not
+# None, are ~ e-14 to e-30 - i.e. 0 to within 13 dp.
+#
+@pytest.mark.parametrize("method,stat,rstat",
+                         [(LevMar, Cash, None),
+                          (LevMar, CStat, pytest.approx(0.0, abs=1e-7)),
+                          (LevMar, Chi2, 0.0),
+                          (NelderMead, Cash, None),
+                          (NelderMead, CStat, 0.0),
+                          (NelderMead, Chi2, 0.0),
+                          (MonCar, Cash, None),
+                          (MonCar, CStat, 0.0),
+                          (MonCar, Chi2, 0.0)])
+def test_fit_dof_1(method, stat, rstat):
+    """DOF is 1"""
+
+    d = Data1D('test', [1, 2, 3], [4, 5, 6], [1, 1, 1])
+    mdl = Polynom1D()
+    mdl.c1.thaw()
+    mdl.c0 = 5.1
+
+    fit = Fit(d, mdl, stat=stat(), method=method())
+    fres = fit.fit()
+
+    assert mdl.c0.val == pytest.approx(3.0, abs=0.001)
+    assert mdl.c1.val == pytest.approx(1.0, abs=0.001)
+
+    assert_fit_results(fres, rstat)
+
+
+# Results from CIAO 4.10 on Linux Ubuntu. Expect to end up with
+# c0 = 3.0, c1 = 1.0, both to about 3dp, and c2 = 0 to within
+# about 1e-6, the rstat values are NaN
+#
+@pytest.mark.parametrize("method,stat,rstat",
+                         [(LevMar, Cash, None),
+                          (LevMar, CStat, np.nan),
+                          (LevMar, Chi2, np.nan),
+                          (NelderMead, Cash, None),
+                          (NelderMead, CStat, np.nan),
+                          (NelderMead, Chi2, np.nan),
+                          (MonCar, Cash, None),
+                          (MonCar, CStat, np.nan),
+                          (MonCar, Chi2, np.nan)])
+def test_fit_dof_0(method, stat, rstat):
+    """DOF is 0"""
+
+    d = Data1D('test', [1, 2, 3], [4, 5, 6], [1, 1, 1])
+    mdl = Polynom1D()
+    mdl.c1.thaw()
+    mdl.c2.thaw()
+    mdl.c0 = 5.1
+
+    fit = Fit(d, mdl, stat=stat(), method=method())
+    fres = fit.fit()
+
+    assert mdl.c0.val == pytest.approx(3.0, abs=0.001)
+    assert mdl.c1.val == pytest.approx(1.0, abs=0.001)
+    assert mdl.c2.val == pytest.approx(0.0, abs=1e-6)
+
+    assert_fit_results(fres, rstat)
+
+
+# Results from CIAO 4.10 on Linux Ubuntu. The fit results are
+# not consistent between different options (e.g. method and statistic),
+# which is not surprising given that the fit is not-well constrained.
+#
+# The only check here is whether the fit succeeds or not, since this
+# depends on the method.
+#
+@pytest.mark.parametrize("stat", [Cash, CStat, Chi2])
+@pytest.mark.parametrize("method,success",
+                         [pytest.param(LevMar, False,
```

Comment:
We need the `pytest.param` part removed, since the test with `LevMar` shouldn't be skipped any more.

Reply:
Done.